build-tools/packages/build-tools/src/type-test-generator/main.ts

```diff
@@ -4,96 +4,54 @@
  */
 
 import { readJsonSync } from "fs-extra";
-import { existsSync, mkdirSync, rmSync, writeFileSync } from "node:fs";
+import { mkdirSync, writeFileSync, rmSync } from "node:fs";
 import path from "node:path";
-import { Project, SourceFile } from "ts-morph";
-import { BrokenCompatTypes } from "../common/fluidRepo";
-import { PackageJson } from "../common/npmPackage";
-import { buildTestCase, TestCaseTypeData } from "../typeValidator/testGeneration";
-import { getFullTypeName, getNodeTypeData, TypeData } from "../typeValidator/typeData";
 import { typeOnly } from "./compatibility";
+import {
+	ensureDevDependencyExists,
+	getPreviousPackageJsonPath,
+	getTypeRollupPathFromExtractorConfig,
+	getTypeDefinitionFilePath,
+	typeDataFromFile,
+	generateCompatibilityTestCases,
+	prepareFilepathForTests,
+	initializeProjectsAndLoadFiles,
+} from "./typeTestUtils";
+import { PackageJson } from "../common/npmPackage";
 
 // Do not check that file exists before opening:
 // Doing so is a time of use vs time of check issue so opening the file could fail anyway.
 // Do not catch error from opening file since the default behavior is fine (exits process with error showing useful message)
 const packageObject: PackageJson = readJsonSync("package.json");
+const previousPackageName = `${packageObject.name}-previous`;
```

Comment:
I love how simple the flow of this script is now! Thanks for all the clean up!!

One remaining recommendation would be to add brief comments above the various function calls explaining what they are doing in terms of the script's progression. That way a reader can quickly correlate the semantic progression with the function calls being made.